velox/type/DecimalUtil.h

```diff
@@ -272,5 +329,87 @@ class DecimalUtil {
   static void toByteArray(int128_t value, char* out, int32_t& length);
 
   static constexpr __uint128_t kOverflowMultiplier = ((__uint128_t)1 << 127);
+
+ private:
+  struct DecimalComponents {
+    std::string_view wholeDigits;
+    std::string_view fractionalDigits;
+    int32_t exponent = 0;
+    char sign = 0;
+    bool hasExponent = false;
+  };
+
+  inline static size_t parseDigitsRun(
+      const char* s,
+      size_t start,
+      size_t size,
+      std::string_view* out) {
+    size_t pos;
+    for (pos = start; pos < size; ++pos) {
+      if (!(s[pos] >= '0' && s[pos] <= '9')) {
+        break;
+      }
+    }
+    *out = std::string_view(s + start, pos - start);
+    return pos;
+  }
+
+  static bool
+  parseDecimalComponents(const char* s, size_t size, DecimalComponents* out) {
+    size_t pos = 0;
+
+    if (size == 0) {
+      return false;
+    }
+    // Sign of the number.
+    if (s[pos] == '-' || s[pos] == '+') {
+      out->sign = *(s + pos);
+      ++pos;
+    }
+    // First run of digits.
+    pos = parseDigitsRun(s, pos, size, &out->wholeDigits);
+    if (pos == size) {
+      return !out->wholeDigits.empty();
+    }
+    // Optional dot (if given in fractional form).
+    if (s[pos] == '.') {
+      // Second run of digits.
+      ++pos;
+      pos = parseDigitsRun(s, pos, size, &out->fractionalDigits);
+    }
+    if (out->wholeDigits.empty() && out->fractionalDigits.empty()) {
+      // Need at least some digits (whole or fractional).
+      return false;
+    }
+    if (pos == size) {
+      return true;
+    }
+    // Optional exponent.
+    if (s[pos] == 'e' || s[pos] == 'E') {
```

Comment:
It would be nice to update the documentation to explain valid formats of the strings that can be casted to decimals.